pkg/operator/encryption/controllers/migrators/metrics.go

```diff
@@ -14,38 +16,50 @@ const (
 var metrics *migratorMetrics
 
 func init() {
-	metrics = newMigratorMetrics()
+	metrics = newMigratorMetrics(legacyregistry.Register)
 }
 
 // migratorMetrics instruments core migrator with prometheus metrics.
 type migratorMetrics struct {
-	objectsMigrated *k8smetrics.CounterVec
-	migration       *k8smetrics.CounterVec
+	objectsMigrated   *k8smetrics.CounterVec
+	migration         *k8smetrics.CounterVec
+	migrationDuration *k8smetrics.HistogramVec
 }
 
 // newMigratorMetrics create a new MigratorMetrics, configured with default metric names.
-func newMigratorMetrics() *migratorMetrics {
+func newMigratorMetrics(registerFunc func(k8smetrics.Registerable) error) *migratorMetrics {
 	objectsMigrated := k8smetrics.NewCounterVec(
 		&k8smetrics.CounterOpts{
 			Namespace: namespace,
 			Subsystem: subsystem,
 			Name:      "migrated_objects",
-			Help:      "The number of objects that have been migrated, labeled with the full resource name.",
+			Help:      "The total number of objects that have been migrated, labeled with the full resource name",
 		}, []string{"resource"})
-	legacyregistry.Register(objectsMigrated)
+	registerFunc(objectsMigrated)
 
 	migration := k8smetrics.NewCounterVec(
 		&k8smetrics.CounterOpts{
 			Namespace: namespace,
 			Subsystem: subsystem,
 			Name:      "migrations",
-			Help:      "The number of completed migration, labeled with the full resource name, and the status of the migration (failed or succeeded)",
+			Help:      "The total number of completed migration, labeled with the full resource name, and the status of the migration (failed or succeeded)",
 		}, []string{"resource", "status"})
-	legacyregistry.Register(migration)
+	registerFunc(migration)
+
+	migrationDuration := k8smetrics.NewHistogramVec(
+		&k8smetrics.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "migration_duration_seconds",
+			Help:      "How long a successful migration takes in seconds, labeled with the full resource name",
+			Buckets:   prometheus.ExponentialBuckets(300, 2, 7),
```

Comment:
actually with this being by resource, should we go down a bit with the minimum? maybe 2m, 4m, 8m, 16m, 32m, 1h, 2h ?

Reply:
yes, it makes sense, updated.